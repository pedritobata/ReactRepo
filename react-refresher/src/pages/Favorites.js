import React, {useContext} from 'react';
import MeetupList from '../components/meetups/MeetupList';
import FavoritesContext from '../store/favorites-context';


const Favorites = () => {
const favoritesCtx = useContext(FavoritesContext);
    return (
        <section>
            <h1>Favorite meetups</h1>
            <MeetupList meetups={favoritesCtx.favoritesMeetups} />
        </section>
    )
}

export default Favorites;