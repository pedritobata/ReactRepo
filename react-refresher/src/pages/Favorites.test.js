import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import Favorites from "./Favorites";
import FavoritesContext from "../store/favorites-context";

const meetupsDummy = [
  {
    id: "m1",
    title: "This is a first meetup",
    image:
      "http://images.com/image1.jpeg",
    address: "Meetupstreet 5, 12345 Lima City",
    description:
      "This is a first, amazing meetup which you definitely should not miss. It will be a lot of fun!",
  },
  {
    id: "m2",
    title: "This is a second meetup",
    image:
      "http://images.com/image2.jpeg",
    address: "Meetupstreet 5, 12345 Chiclayo City",
    description:
      "This is a second, amazing meetup which you definitely should not miss. It will be a lot of fun!",
  },
];

describe("Favorites Screen", () => {
  it("should render correctly", () => {
    const snapshot = renderer.create(<Favorites />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('should render a list of favorite meetups', () => {
      const mockContext = {
        favoritesMeetups: meetupsDummy,
        favotitesTotal: meetupsDummy.length,
        addFavorite: () => {},
        removeFavorite: () => {},
        isFavorite: () => {},
      } 

      render(<FavoritesContext.Provider value={mockContext}><Favorites /></FavoritesContext.Provider>);
       
      const meetupsHeadings = screen.getAllByRole('heading');
      expect(meetupsHeadings[0]).toHaveTextContent('Favorite meetups');
      expect(meetupsHeadings[1]).toHaveTextContent(meetupsDummy[0].title);
      expect(meetupsHeadings[2]).toHaveTextContent(meetupsDummy[1].title);
  })
});
