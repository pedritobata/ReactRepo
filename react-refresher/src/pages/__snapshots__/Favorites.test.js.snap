// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites Screen should render correctly 1`] = `
<section>
  <h1>
    Favorite meetups
  </h1>
  <ul
    className="list"
  />
</section>
`;
